src/components/AnalyticsSearch/index.tsx

```diff
@@ -0,0 +1,361 @@
+              });
+              return !included;
+            }),
+          );
+
+          const foundTokens = tokens.data.asSymbol
+            .concat(tokens.data.asAddress)
+            .concat(tokens.data.asName);
+
+          allTokens = allTokens.concat(
+            foundTokens.filter((searchedToken: any) => {
+              let included = false;
+              allTokens.map((token) => {
+                if (token.id === searchedToken.id) {
+                  included = true;
+                }
+                return true;
+              });
+              return !included;
+            }),
+          );
+        }
+
+        const foundTokensWithData = await Promise.all(
+          allTokens.map(async (token: any) => {
+            const utcCurrentTime = dayjs();
+            const utcOneDayBack = utcCurrentTime.subtract(1, 'day').unix();
+            const oneDayBlock = await getBlockFromTimestamp(utcOneDayBack);
+            const oneDayResult = await client.query({
+              query: TOKEN_INFO_OLD(oneDayBlock, token.id),
+              fetchPolicy: 'network-only',
+            });
+            if (
+              oneDayResult &&
+              oneDayResult.data &&
+              oneDayResult.data.tokens &&
+              oneDayResult.data.tokens.length > 0
+            ) {
+              const oneDayHistory = oneDayResult.data.tokens[0];
+              const oneDayVolumeUSD =
+                (token?.tradeVolumeUSD ?? 0) -
+                (oneDayHistory?.tradeVolumeUSD ?? 0);
+              return { ...token, oneDayVolumeUSD };
+            }
+            return token;
+          }),
+        );
+        setSearchedTokens(foundTokensWithData);
+        setSearchedPairs(allPairs);
+      } catch (e) {
+        console.log(e);
+      }
+    }
+    fetchData();
+  }, [searchVal]);
+
+  const handleClick = (e: any) => {
+    if (
+      !(menuRef.current && menuRef.current.contains(e.target)) &&
+      !(wrapperRef.current && wrapperRef.current.contains(e.target))
+    ) {
+      setPairsShown(3);
+      setTokensShown(3);
+      setMenuOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  }, []);
+
+  return (
+    <Box className='searchWidgetWrapper'>
+      <Box className='searchWidgetInput'>
+        <input
+          placeholder={t('searchTokenPair')}
+          value={searchValInput}
+          ref={menuRef}
+          onFocus={() => setMenuOpen(true)}
+          onChange={(evt) => setSearchValInput(evt.target.value)}
+        />
+        <Box display='flex'>
+          <SearchIcon />
+        </Box>
+      </Box>
+      {menuOpen && (
+        <div ref={wrapperRef} className='searchWidgetContent'>
+          <h6>{t('pairs')}</h6>
+          {filteredPairs.slice(0, pairsShown).map((val, ind) => {
+            const currency0 = new Token(
+              ChainId.MATIC,
```

Comment:
store the chain id in a const and add a comment with a todo to support multichain

Reply:
Fixed the issue and pushed a commit